presto-hive-metastore/src/main/java/com/facebook/presto/hive/metastore/HiveMetastoreCacheStats.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.hive.metastore;
+
+import com.facebook.presto.hive.metastore.CachingHiveMetastore.KeyAndContext;
+import com.google.common.cache.Cache;
+import com.google.common.cache.LoadingCache;
+import org.weakref.jmx.Managed;
+
+import java.util.Optional;
+
+public class HiveMetastoreCacheStats
+{
+    private Optional<LoadingCache<KeyAndContext<HivePartitionName>, Optional<Partition>>> partitionCache = Optional.empty();
+
+    public void setPartitionCache(LoadingCache<KeyAndContext<HivePartitionName>, Optional<Partition>> partitionCache)
+    {
+        this.partitionCache = Optional.of(partitionCache);
+    }
```

Comment:
not a big fan of using setter here. But it's also hard to inject given the cache creation requires overloaded `load` methods. Any good suggestion @rschlussel ?